### ISSUE TITLE
Configmap properties changes are not being detected in dev mode

### ISSUE THREAD

**Issue by ammachado** [apache/camel-k#3347]
I'm using `kamel run --dev` to test my integration.  All my configurations are on a config map on the cluster.

My integration declares a config `modeline` using a configmap.

The first time I run `kamel run --dev Integration.java` the configuration is loaded from the configmap.  I can change the code and changes are reflected on the integration without any issue.

However, if I make any change on the configmap and try to either reload the integration by making a code change or by hitting `Ctrl+C` and running `kamel run --dev Integration.java` again, the changes on the configmap are not reflected in the integration.

The only way to reflect the changes is either create a new configmap and change the modeline to the new configmap or deleting everything (the integration and the integrationkit) from the cluster.

Is this the expected behavior?

When `kamel run --dev Integration.java` is running a pod is created on the cluster.  For the sake of curiosity, I inspected the YAML of the pod and there isn't a volume mount for the configmap, but it is present when you run the integration without the `--dev` parameter.

**Comment by squakez** [comment]
It's a known behavior. There is a feature request for it #1235

**Comment by squakez** [comment]
Just tested this successfully. I've created an Integration with a modeline configmap, and run it, having:
```
....
[1] 2022-07-13 11:02:57,477 INFO  [configmap] (Camel (camel-1) thread #1 - timer://configmap) configmap content is: configmap content
...
```
In parallel I've changed the content of the configmap, and, as expected, the Integration won't automatically pick those new values (no hot deploy on configmap changes). However, stopping and restarting the application, triggered the change as expected:
```
....
[1] 2022-07-13 11:03:24,448 INFO  [configmap] (Camel (camel-1) thread #1 - timer://configmap) configmap content is: configmap content 2
....
```
> When kamel run --dev Integration.java is running a pod is created on the cluster. For the sake of curiosity, I inspected the YAML of the pod and there isn't a volume mount for the configmap, but it is present when you run the integration without the --dev parameter.

The `Pod` spec should be exactly the same. I've just checked in the `--dev` mode and it mounts the configmap as expected.

I'm closing this in favor of the feature request for configmap hot deploy.
